src/hotspot/share/opto/predicates.hpp

```diff
@@ -369,6 +369,12 @@ class RuntimePredicate : public Predicate {
   static bool is_predicate(Node* node, Deoptimization::DeoptReason deopt_reason);
 };
 
+// Utility class to represent an Assertion Predicate.
+class AssertionPredicate : public StackObj {
+ public:
+  static bool may_be_predicate_if(const Node* node);
+};
+
```

Comment:
Do you really need separate class for one static method? Can the method be a local static in `predicates.cpp` file?